homeassistant/components/tado/entity.py

```diff
@@ -17,16 +17,23 @@ def __init__(self, device_info: dict[str, str]) -> None:
         """Initialize a Tado device."""
         super().__init__()
         self._device_info = device_info
-        self.device_name = device_info["serialNo"]
-        self.device_id = device_info["shortSerialNo"]
+        self.device_name = device_info.get("serialNo", device_info.get("serialNumber"))
```

Comment:
This seems impactful and not backwards compatible. Current devices already mapped, we be added as a new device. Why is this needed?

Reply:
TadoX doesn't provide a short serial number and the serial number isn't called serialNo anymore but serialNumber. It should not change the device id or device name for not X devices since array.get(key, default) basically does the same as array[key] but doesn't throw an error if the key doesn't exist, it uses the default instead which for TadoX is serialNumber.

Luckily I have access to a tado v3 system and can test it but I'm pretty sure it's no problem.

Reply:
Good to know to it doesn't interfere with current installations. I don't have a TadoX so I miss some context. Thanks for clarifying! :)

Reply:
I've changed that part so it becomes more readable. It doesn't interfere with current installations.